### PR TITLE
Added java epsilon test for double values [skip ci]

### PR DESCRIPTION
**Pull request by razajafri**
This addresses a problem when comparing double values that are big, the difference between the actual and expected is large but still relatively small in value compared to expected

### CODE REVIEW

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java [hunk 1]
```diff
@@ -647,7 +647,7 @@ void testQuantilesOnIntegerInput() {
       for (int j = 0 ; j < quantiles.length ; j++) {
         for (int i = 0 ; i < methods.length ; i++) {
           try(Scalar result = cv.quantile(methods[i], quantiles[j])) {
-            assertEquals(exactExpected[i][j], result.getDouble(), DELTA);
+            assertEqualsWithEpsilon(exactExpected[i][j], result.getDouble(), DELTA);
```
Comment (revans2) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java]: So is this an Epsilon or a delta?  I am not sure that using greek letters is clear enough to denote what we want. I would almost rather see it be explicit like `assertEqualsWithinPercentage`.

File: java/src/test/java/ai/rapids/cudf/CudfTestBase.java [hunk 2]
```diff
@@ -108,4 +109,35 @@ static void afterAll() {
     }
     LISTENER.checkOutstanding();
   }
+
+  private static boolean doublesAreEqualWithEpsilon(double expected, double actual, double epsilon) {
+    if (Double.doubleToLongBits(expected) != Double.doubleToLongBits(actual)) {
```
Comment (revans2) [inline java/src/test/java/ai/rapids/cudf/CudfTestBase.java]: I don't see the value in this check.  It is extra code that does not appear to add any value.

**Comment by jrhemstad** [comment]
Drive by comment. Comparing floats with a fixed episilon is bad. Instead, use units in the last place. See https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/

File: java/src/test/java/ai/rapids/cudf/CudfTestBase.java (hunk 2 above)
Comment (razajafri) [reply java/src/test/java/ai/rapids/cudf/CudfTestBase.java]: We need that check for equality test of NaNs, +ve infin and -ve infin. I have documented the code to be explicit

**Comment by razajafri** [comment]
@jrhemstad thanks for the comment. I have read the article and it doesn't look like it will be very straight forward to do this. I have created an [issue ](https://github.com/rapidsai/cudf/issues/4383) to track this.

**Comment by razajafri** [comment]
@revans2 I have addressed your concerns I think PTAL

**Comment by harrism** [comment]
What is the status of this for 0.13? Please remember to add PRs to a project.

image

**Comment by razajafri** [comment]
This isn't critical for 0.13
We can either check it in 0.13 or 0.14

**Comment by revans2** [comment]
Let's just move it to 0.14.

**Comment by razajafri** [comment]
@kkraus14 can we get this in?

File: CHANGELOG.md [hunk 3]
```diff
@@ -107,6 +107,7 @@
 
 ## Improvements
 
+- PR #4331 Improved test for double that considers an epsilon
```
Comment (kkraus14) [inline CHANGELOG.md]: This is still in 0.13 changelog

Comment (razajafri) [reply CHANGELOG.md]: I have addressed your concern. Can you please check?

**Pull request merged**